runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +426,41 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.showShortcutLevel())
+		{
+			return;
+		}
+
+		//Guarding against non-first option because agility shortcuts are always that type of event.
+		if (event.getType() != MenuAction.GAME_OBJECT_FIRST_OPTION.getId())
+		{
+			return;
+		}
+
+		final int entryId = event.getIdentifier();
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		for (Obstacle nearbyObstacle : getObstacles().values())
+		{
+			AgilityShortcut shortcut = nearbyObstacle.getShortcut();
+			if (Arrays.stream(shortcut.getObstacleIds()).anyMatch(i -> i == entryId))
+			{
+				MenuEntry entry = menuEntries[menuEntries.length - 1];
+				if (!entry.getTarget().toLowerCase().contains("(level-"))
```

Comment:
what is this check for

Reply:
^ agree doesn't seem to be necessary unless there's an edge case we aren't considering

Reply:
True, the "level-" check should never actually crop up. I think that was a check left over from when I was doing things in a much different way that I neglected to remove.